modules/http4s/test/src/smithy4s/http4s/SparseQueryParametersSuite.scala

```diff
@@ -0,0 +1,109 @@
+package smithy4s.http4s
+
+import cats.effect.kernel.Deferred
+import cats.effect.IO
+import io.circe.Json
+import org.http4s._
+import org.http4s.circe.CirceInstances
+import org.http4s.client.Client
+import org.http4s.implicits._
+
+import smithy4s.example.{ServiceWithSparseQueryParams, SparseQueryOutput}
+import weaver._
+
+object SparseQueryParametersSuite extends SimpleIOSuite with CirceInstances {
```

Comment:
I'd rather not have a bespoke http4s test for this. I'd rather the test was encoded as a compliance test in alloy instead, but that could happen in later PR

Reply:
I tried adding a compliance test in https://github.com/disneystreaming/alloy/pull/172 , but for some reason that I can't comprehend it fails:

```sbt
[info] [http4s] smithy4s.http4s.SimpleRestJsonComplianceSuite
...
[info] [http4s] + alloy.test#SparseQueryParam(client|Request): SparseQueryParam 676ms
[info] [http4s] - alloy.test#SparseQueryParam(server|Request): SparseQueryParam 202ms
[info] [http4s] *************FAILURES**************
[info] [http4s] smithy4s.http4s.SimpleRestJsonComplianceSuite
[error] [http4s] - alloy.test#SparseQueryParam(server|Request): SparseQueryParam 202ms
[error] [http4s]    the result value: Array(List(Some(value = "bar"), Some(value = ""), Some(value = "baz"), Some(value = ""))) was not equal to the expected TestCase value Array(List(Some(value = "bar"), None, Some(value = "baz"), Some(value = ""))). (modules/tests/src/smithy4s/tests/ProtocolComplianceSuite.scala:183)
```

The client part passes, but the server one doesn't and I just can't wrap my head around why 🤯